web/src/presence.ts

```diff
@@ -304,6 +322,48 @@ export function last_active_date(user_id: number): Date | undefined {
     return new Date(info.last_active * 1000);
 }
 
+export function map_status(status: string): "active" | "idle" | "offline" {
+    switch (status) {
+        case "active":
+        case "idle":
+        case "offline":
+            return status;
+        default:
+            return "offline";
+    }
+}
+
+export const fetch_user_last_seen_status = (user_id: number, callback: () => void): void => {
+    const url = `json/users/${user_id}/presence`;
+
+    channel.get({
+        url,
+        success(data: unknown) {
+            const parsed_data = user_last_seen_response_schema.safeParse(data);
+
+            if (parsed_data.success) {
+                const response = parsed_data.data;
+
+                if (response.result === "success" && response.presence) {
+                    const {aggregated} = response.presence;
+                    presence_info.set(user_id, {
+                        status: map_status(aggregated.status),
+                        last_active: aggregated.timestamp,
+                    });
+                } else {
+                    // fallback logic for users who haven't generated any presence data
+                    const user = people.get_by_user_id(user_id);
+                    presence_info.set(user_id, {
+                        status: "offline",
+                        last_active: new Date(user.date_joined).getTime() / 1000,
+                    });
+                }
+            }
+            callback();
+        },
+    });
+};
```

Comment:
This function was losing 100% test coverage in `presence.test.cjs`, so I added this file to `test-js-with-node`. I'm not entirely sure whether I should try to write a test for this or not. I looked at various files where `channel.get` was used, and most of those files were included in `test-js-with-node` due to the loss of 100% test coverage.